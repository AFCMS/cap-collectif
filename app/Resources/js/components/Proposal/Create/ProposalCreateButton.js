// @flow
import React from 'react';
import { FormattedMessage } from 'react-intl';
import { Button } from 'react-bootstrap';
import {createFragmentContainer, graphql} from "react-relay";
import LoginOverlay from '../../Utils/LoginOverlay';
import type {ProposalCreateButton_proposalForm} from "./__generated__/ProposalCreateButton_proposalForm.graphql";

type Props = {
  handleClick: Function,
  disabled: boolean,
  proposalForm: ProposalCreateButton_proposalForm
};

export class ProposalCreateButton extends React.Component<Props> {
  render() {
    const { disabled, handleClick, proposalForm } = this.props;
    const buttonTradKey = proposalForm.isProposalForm ? "proposal.add" : " submit-a-question";
    return (
      <LoginOverlay>
        <Button
          id="add-proposal"
          disabled={disabled}
          bsStyle="primary"
          onClick={disabled ? null : handleClick}>
          <i className="cap cap-add-1" />
          <FormattedMessage id={buttonTradKey} />
        </Button>
      </LoginOverlay>
    );
  }
}

export default createFragmentContainer(ProposalCreateButton, {
  proposalForm: graphql`
    fragment ProposalCreateButton_proposalForm on ProposalForm {
      id
      isProposalForm
    }
  `,
});
