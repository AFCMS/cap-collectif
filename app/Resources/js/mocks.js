// @flow
/* eslint-env jest */
import { type FormProps } from 'redux-form';
import { type IntlShape } from 'react-intl';
import type { RelayPaginationProp, RelayRefetchProp } from 'react-relay';

export const $refType: any = null;
export const $fragmentRefs: any = null;

export const intlMock: IntlShape = {
  locale: 'fr-FR',
  formats: {},
  messages: {},
  now: () => 0,
  formatHTMLMessage: message => String(message),
  formatPlural: message => String(message),
  formatNumber: message => String(message),
  formatRelative: message => String(message),
  formatTime: message => String(message),
  formatDate: message => String(message),
  formatMessage: message => String(message.id),
};

export const formMock: FormProps = {
  anyTouched: false,
  array: {
    insert: jest.fn(),
    move: jest.fn(),
    pop: jest.fn(),
    push: jest.fn(),
    remove: jest.fn(),
    removeAll: jest.fn(),
    shift: jest.fn(),
    splice: jest.fn(),
    swap: jest.fn(),
    unshift: jest.fn(),
  },
  asyncValidate: jest.fn(),
  asyncValidating: false,
  autofill: jest.fn(),
  blur: jest.fn(),
  change: jest.fn(),
  clearAsyncError: jest.fn(),
  clearSubmit: jest.fn(),
  destroy: jest.fn(),
  dirty: false,
  dispatch: jest.fn(),
  error: null,
  form: 'formName',
  handleSubmit: jest.fn(),
  initialize: jest.fn(),
  initialized: true,
  initialValues: {},
  invalid: false,
  pristine: true,
  reset: jest.fn(),
  submitting: false,
  submitFailed: false,
  submitSucceeded: false,
  touch: jest.fn(),
  untouch: jest.fn(),
  valid: true,
  warning: null,
};

export const relayPaginationMock: RelayPaginationProp = {
  environment: {
    applyMutation: jest.fn(),
    sendMutation: jest.fn(),
    lookup: jest.fn(),
    sendQuery: jest.fn(),
    subscribe: jest.fn(),
    streamQuery: jest.fn(),
    retain: jest.fn(),
    unstable_internal: {},
  },
  hasMore: () => false,
  isLoading: () => false,
  loadMore: jest.fn(),
  refetchConnection: jest.fn(),
};

export const relayRefetchMock: RelayRefetchProp = {
  environment: {
    applyMutation: jest.fn(),
    sendMutation: jest.fn(),
    lookup: jest.fn(),
    sendQuery: jest.fn(),
    subscribe: jest.fn(),
    streamQuery: jest.fn(),
    retain: jest.fn(),
    unstable_internal: {},
  },
  refetch: jest.fn(),
};
