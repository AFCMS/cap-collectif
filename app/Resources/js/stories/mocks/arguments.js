// @flow
import { author } from './users';

export const argument = {
  user: author,
  trashedStatus: '',
  createdAt: '1 mars 2016 à 00:00',
  publishedAt: '1 mars 2016 à 00:00',
  body:
    'In placeat reiciendis ut. Officiis praesentium quia minima ut tenetur officiis. Eaque fugit voluptates temporibus suscipit provident culpa culpa. Magni recusandae dolorem aut id.',
  votes: { totalCount: 50 },
};

export const argumentsData = [
  { ...argument, user: { ...argument.user, vip: true } },
  { ...argument },
  { ...argument, trashedStatus: 'INVISIBLE' },
  { ...argument },
];
