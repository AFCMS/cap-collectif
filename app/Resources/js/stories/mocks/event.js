// @flow
import { author } from './users';

export const event = {
  title: 'PHPTourDuTurfu',
  url: '#',
  image: {
    url: 'https://picsum.photos/300/400',
    alt: 'event picture',
  },
  date: {
    start: '2030-03-10 00:00:00',
    end: '2032-06-15 00:00:00',
  },
  location: {
    place: 'Tour Eiffel',
    zip: '75007',
    city: 'Paris',
  },
  category: 'Immobilier',

  user: author,
};
