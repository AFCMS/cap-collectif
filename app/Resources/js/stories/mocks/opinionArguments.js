// @flow
import { author } from './users';

export const opinionArgument = {
  user: author,
  trashedStatus: '',
  createdAt: '1 mars 2016 à 00:00',
  publishedAt: '1 mars 2016 à 00:00',
  body:
    'In placeat reiciendis ut. Officiis praesentium quia minima ut tenetur officiis. Eaque fugit voluptates temporibus suscipit provident culpa culpa. Magni recusandae dolorem aut id.',
  votes: { totalCount: 50 },
};

export const opinionArguments = [
  { ...opinionArgument, user: { ...opinionArgument.user, vip: true } },
  { ...opinionArgument },
  { ...opinionArgument, trashedStatus: 'INVISIBLE' },
  { ...opinionArgument },
];
