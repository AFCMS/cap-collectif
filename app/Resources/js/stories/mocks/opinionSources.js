// @flow
import { author } from './users';

export const opinionSource = {
  url: 'https://ui.cap-collectif.com/',
  user: author,
  createdAt: ' • 1 mars 2016',
  publishedAt: ' • 1 mars 2016',
  title: 'Source title',
  category: 'Politique',
  body:
    'In placeat reiciendis ut. Officiis praesentium quia minima ut tenetur officiis. Eaque fugit voluptates temporibus suscipit provident culpa culpa. Magni recusandae dolorem aut id.',
  votes: { totalCount: 2 },
};

export const opinionSources = [
  { ...opinionSource },
  { ...opinionSource },
  { ...opinionSource, user: { ...opinionSource.user, vip: true } },
  { ...opinionSource },
  { ...opinionSource, category: null },
];
