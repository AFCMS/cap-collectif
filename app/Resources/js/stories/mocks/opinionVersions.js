// @flow
import { author } from './users';

export const opinionVersion = {
  url: 'https://ui.cap-collectif.com/',
  title: 'Opinion version',
  user: author,
  createdAt: ' • 3 janvier 2015',
  pinned: true,
  publishedAt: ' • 3 janvier 2015',
  votes: { totalCount: 50 },
  arguments: { totalCount: 4 },
  sources: { totalCount: 0 },
  votesMitige: { totalCount: 1 },
  votesNok: { totalCount: 1 },
  votesOk: { totalCount: 2 },
};

export const opinionVersions = [
  { ...opinionVersion },
  { ...opinionVersion, user: { ...opinionVersion.user, vip: true } },
  { ...opinionVersion, votes: { totalCount: 0 } },
];
