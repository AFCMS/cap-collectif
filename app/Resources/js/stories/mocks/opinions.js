// @flow
import { author } from './users';

export const opinion = {
  url: 'https://ui.cap-collectif.com/',
  title: 'Opinion',
  user: author,
  createdAt: ' • 1 mars 2018',
  pinned: true,
  publishedAt: ' • 1 mars 2018',
  votes: { totalCount: 4 },
  versions: { totalCount: 3 },
  arguments: { totalCount: 4 },
  sources: { totalCount: 0 },
  votesMitige: { totalCount: 1 },
  votesNok: { totalCount: 1 },
  votesOk: { totalCount: 2 },
};

export const opinions = [
  { ...opinion },
  { ...opinion, user: { ...opinion.user, vip: true } },
  { ...opinion, votes: { totalCount: 0 } },
];
