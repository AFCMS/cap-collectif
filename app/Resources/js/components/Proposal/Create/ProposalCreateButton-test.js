// @flow
/* eslint-env jest */
import * as React from 'react';
import {shallow} from 'enzyme';
import {ProposalCreateButton} from './ProposalCreateButton';
import {$refType, $fragmentRefs} from '../../../mocks';

describe('<ProposalCreateButton />', () => {

  it('should render correctly', () => {
    const props = {
      handleClick: jest.fn(),
      disabled: false,
      // $FlowFixMe $refType
      proposalForm: {
        $refType,
        $fragmentRefs,
        isProposalForm: true
      }
    };
    const wrapper = shallow(<ProposalCreateButton {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render correctly when disabled', () => {
    const props = {
      handleClick: jest.fn(),
      // $FlowFixMe $refType
      proposalForm: {
        $refType,
        $fragmentRefs,
        isProposalForm: true
      },
      disabled: true,
    };
    const wrapper = shallow(<ProposalCreateButton {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render correctly as Question', () => {
    const props = {
      handleClick: jest.fn(),
      disabled: false,
      // $FlowFixMe $refType
      proposalForm: {
        $refType,
        $fragmentRefs,
        isProposalForm: false
      }
    };
    const wrapper = shallow(<ProposalCreateButton {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
