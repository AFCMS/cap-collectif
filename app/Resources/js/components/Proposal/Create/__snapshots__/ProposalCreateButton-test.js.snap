// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProposalCreateButton /> should render correctly 1`] = `
<Connect(LoginOverlay)>
  <Button
    active={false}
    block={false}
    bsClass="btn"
    bsStyle="primary"
    disabled={false}
    id="add-proposal"
    onClick={[MockFunction]}
  >
    <i
      className="cap cap-add-1"
    />
    <FormattedMessage
      id="proposal.add"
      values={Object {}}
    />
  </Button>
</Connect(LoginOverlay)>
`;

exports[`<ProposalCreateButton /> should render correctly as Question 1`] = `
<Connect(LoginOverlay)>
  <Button
    active={false}
    block={false}
    bsClass="btn"
    bsStyle="primary"
    disabled={false}
    id="add-proposal"
    onClick={[MockFunction]}
  >
    <i
      className="cap cap-add-1"
    />
    <FormattedMessage
      id=" submit-a-question"
      values={Object {}}
    />
  </Button>
</Connect(LoginOverlay)>
`;

exports[`<ProposalCreateButton /> should render correctly when disabled 1`] = `
<Connect(LoginOverlay)>
  <Button
    active={false}
    block={false}
    bsClass="btn"
    bsStyle="primary"
    disabled={true}
    id="add-proposal"
    onClick={null}
  >
    <i
      className="cap cap-add-1"
    />
    <FormattedMessage
      id="proposal.add"
      values={Object {}}
    />
  </Button>
</Connect(LoginOverlay)>
`;
